### PR TITLE
feat: setup MemCacheStore

### PR DESCRIPTION
**Pull request by jiacai2050**
# Which issue does this PR close?

Closes #

# Rationale for this change
 
After #415, MemCachedStore is implemented, but not used, this PR add it to setup

# What changes are included in this PR?

1. Remove old data cache
2. Add new mem cache options

# Are there any user-facing changes?

No

# How does this change test

Start server locally with config below

```
bind_addr = "127.0.0.1"
http_port = 5440
grpc_port = 8831
log_level = "debug"

[analytic.storage]
mem_cache_capacity = '1G'
mem_cache_partition_bits = 0

[analytic.storage.object_store]
type = "Local"
data_path = "/tmp/xx"

[analytic]
wal_path = "/tmp/xx"
sst_meta_cache_cap = 10000
scan_batch_size = 10000

[[static_route.topology.schema_shards]]
schema = 'public'
auto_create_tables = true

[[static_route.topology.schema_shards.shard_views]]
shard_id = 0

[static_route.topology.schema_shards.shard_views.endpoint]
addr = "127.0.0.1"
port = 8831
```

### CODE REVIEW

File: analytic_engine/src/storage_options.rs [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 
+use common_util::config::ReadableSize;
 use object_store::cache::CachedStoreConfig;
 use serde::Deserialize;
 
+#[derive(Debug, Clone, Deserialize)]
 /// Options for storage backend
+pub struct StorageOptions {
+    pub mem_cache_capacity: ReadableSize,
+    pub mem_cache_partition_bits: usize,
```
Comment (Rachelint) [inline analytic_engine/src/storage_options.rs:11]: Should we expose cache informations here?

File: analytic_engine/src/setup.rs [hunk 2]
```diff
@@ -270,22 +264,34 @@ fn open_storage(
...
                 let remote_store = open_storage(*cache_opts.remote_store).await?;
                 let store = CachedStore::init(local_store, remote_store, cache_opts.cache_opts)
                     .await
                     .context(OpenObjectStore)?;
-                Ok(Arc::new(store) as _)
+                Arc::new(store) as _
             }
+        };
+
+        if opts.mem_cache_capacity.as_bytes() == 0 {
```
Comment (Rachelint) [inline analytic_engine/src/setup.rs:285]: The method name `as_bytes` is confused...

File: analytic_engine/src/storage_options.rs (hunk 1 above)
Comment (jiacai2050) [reply analytic_engine/src/storage_options.rs:11]: Yeah, this config set how many partition we want in cache.

File: analytic_engine/src/setup.rs (hunk 2 above)
Comment (jiacai2050) [reply analytic_engine/src/setup.rs:403]: Agree, `as_byte` may better match as method like `as_mb()`

Comment (jiacai2050) [reply analytic_engine/src/setup.rs:403]: https://github.com/CeresDB/ceresdb/issues/428

**Review by Rachelint — APPROVED** [review]
LGTM.

**Pull request merged**